src/storage/redis_db.cc

```diff
@@ -575,8 +576,10 @@ rocksdb::Status SubKeyScanner::Scan(engine::Context &ctx, RedisType type, const
     }
     InternalKey ikey(iter->key(), storage_->IsSlotIdEncoded());
     keys->emplace_back(ikey.GetSubKey().ToString());
-    if (values != nullptr) {
-      values->emplace_back(iter->value().ToString());
+    if (!no_values_) {
+      if (values != nullptr) {
+        values->emplace_back(iter->value().ToString());
+      }
     }
```

Comment:
`no_values_` seems useless. We can just pass a `nullptr` as `values` to indicate that we don't want values.